spring-core/src/main/java/org/springframework/util/MimeTypeUtils.java

```diff
@@ -181,7 +181,7 @@ public abstract class MimeTypeUtils {
 
 	static {
 		// Not using "parseMimeType" to avoid static init cost
-		ALL = new MimeType("*", "*");
+		ALL = new MimeType(MimeType.WILDCARD_TYPE, MimeType.WILDCARD_TYPE);
```

Comment:
This is just using constants instead of String literals, and it should be invoked only once as a static initializer. I don't see how this improves performance.

Reply:
It allows the reference check to work https://github.com/openjdk/jdk/blob/jdk-17%2B35/src/java.base/share/classes/java/lang/String.java#L1822